.github/workflows/rust.yml

```diff
@@ -15,25 +15,34 @@ jobs:
         uses: actions/checkout@v2
 
       - name: Set up cargo/rust
-        run: |
-          rustup component add rustfmt
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rustfmt, clippy
+          override: true
 
       - name: Setup OCaml (because of ocaml-gen)
-        run: sudo apt update && sudo apt install ocaml
+        run: |
+          sudo apt update
+          sudo apt install -y ocaml
 
-      - name: build
+      # `cargo check` compiles the packages without performing the final step of code generation,
+      # which is faster than running `cargo build`.
+      # Use release build to speed up tests
+      - name: build & test
         run: |
-          cargo build
+          cargo check --release
+          cargo test --release
```

Comment:
can you remove `--release` for tests? As it disable some checks like integer overflows

Reply:
just read your message above, that's a shame :( I guess let's leave the release then

Reply:
Is integer overflow checks the only thing to be concerned?  I believe it can be enabled via [rustc flags](https://doc.rust-lang.org/cargo/reference/profiles.html#overflow-checks)

Reply:
I'm actually not sure what is getting disabled in release mode. But that could be a good compromise!

Reply:
> could be a good compromise!

I will add the `RUSTFLAGS` environment variable then : )

The tests take minutes in debug mode, and I see many warnings like 
```
test bench::tests::test_bench has been running for over 60 seconds
test tests::chacha::chacha_prover has been running for over 60 seconds
```
BTW, previous `cargo test` in the circleci was also under release mode

Reply:
> not sure what is getting disabled

According to the [doc](https://doc.rust-lang.org/cargo/reference/profiles.html#release)
```
debug-assertions = false
overflow-checks = false
```

Reply:
ah, it'd be good to have debug assertions as well!

Reply:
Added `-Cdebug-assertions=y`, also enabled incremental build by setting `CARGO_INCREMENTAL` to `true`